### PR TITLE
String and encoding clean up

### PR DESCRIPTION
**Pull request by nirvdrum**
There's a lot more that can be cleaned up, but this is a start.

### CODE REVIEW

File: src/main/java/org/truffleruby/core/string/StringNodes.java [hunk 1]
```diff
@@ -2118,22 +2118,28 @@ public boolean isStringSubclass(DynamicObject string) {
     }
 
     @CoreMethod(names = {"to_sym", "intern"})
-    @ImportStatic({ StringCachingGuards.class, StringOperations.class })
+    @ImportStatic({ StringCachingGuards.class, StringGuards.class, StringOperations.class })
     public abstract static class ToSymNode extends CoreMethodArrayArgumentsNode {
 
-        @Specialization(guards = "equalNode.execute(rope(string), cachedRope)", limit = "getLimit()")
+        @Specialization(guards = { "!isBrokenCodeRange(string)", "equalNode.execute(rope(string), cachedRope)" },
+                limit = "getLimit()")
```
Comment (eregon) [inline src/main/java/org/truffleruby/core/string/StringNodes.java]: Style suggestion: in similar cases, I break the line after `guards = {` so the guards are on their own line(s).
And then the annotation ends on a new line with
`}, limit = "getLimit()")`

Comment (nirvdrum) [reply src/main/java/org/truffleruby/core/string/StringNodes.java]: Fixed.

File: src/main/java/org/truffleruby/core/encoding/EncodingManager.java [hunk 2]
```diff
@@ -179,21 +176,7 @@ public synchronized DynamicObject replicateEncoding(Encoding encoding, String na
...
     public static Charset charsetForEncoding(Encoding encoding) {
-        final String encodingName = encoding.toString();
-
-        if (encodingName.equals("ASCII-8BIT")) {
-            return StandardCharsets.ISO_8859_1;
-        }
-
-        if (encoding == ISO8859_16Encoding.INSTANCE) {
-            return ISO_8859_16.INSTANCE;
-        }
```
Comment (eregon) [inline src/main/java/org/truffleruby/core/encoding/EncodingManager.java]: Does `encoding.getCharset()` also handle ISO_8859_16 and ASCII-8BIT?

Comment (nirvdrum) [reply src/main/java/org/truffleruby/core/encoding/EncodingManager.java]: `ASCII-8BIT` is handled. But I missed that we had a custom charset for `ISO_8859_16`, so I'll have to investigate that a bit more.

Comment (nirvdrum) [reply src/main/java/org/truffleruby/core/encoding/EncodingManager.java]: `ISO_8859_16` is handled in jcodings as well. jcodings sets up a Charset SPI and registers an `ISO_8859_16` Charset. So, we can remove our local copy of that class. Thanks for pointing that out.

**Pull request merged**